### PR TITLE
Implement OpenSSL::BN#abs

### PR DESCRIPTION
**Pull request by rickmark**
Provides an efficient way to guarantee positive values

Resolves #429 - missing chunk from prior PR

### CODE REVIEW

File: ext/openssl/ossl_bn.c [hunk 1]
```diff
@@ -960,6 +960,29 @@ ossl_bn_uminus(VALUE self)
...
+    if (BN_is_negative(bn1)) {
+        obj = NewBN(cBN);
+        bn2 = BN_dup(bn1);
+        if (!bn2) { ossl_raise(eBNError, "BN_dup"); }
+        SetBN(obj, bn2);
+        BN_set_negative(bn2, 0);
+        return obj;
+    } else {
+        return self;
+    }
```
Comment (rhenium) [inline ext/openssl/ossl_bn.c:983]: `OpenSSL::BN` is a mutable object (`#set_bit!` exists, for example), so it should create a new instance of `OpenSSL::BN` even if the value is already positive.

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,5 @@
 ext/openssl/mkmf.log
 ext/openssl/Makefile
 ext/openssl/extconf.h
+.DS_Store
+.idea/
```
Comment (rhenium) [inline .gitignore:12]: Unrelated changes checked in?

Comment (rickmark) [reply .gitignore:12]: It's never a change by itself but removing Intelli-J and macOS files from being tracked is something I do with first contribution.  Some projects find it helpful, some do not

File: ext/openssl/ossl_bn.c (hunk 1 above)
Comment (rickmark) [reply ext/openssl/ossl_bn.c:983]: I was trying to track the behavior of unary plus.  I figured they should be the same, unary plus returns the same object, unary minus is always a copy.  abs did whatever was appropriate.  I know you're trying to imply that there are a ton of subtle bugs that come from "mutable primitives" which a BN still feels pretty Int like when you're using it for math....

File: test/openssl/test_bn.rb [hunk 3]
```diff
@@ -114,6 +114,19 @@ def test_sqr
     assert_equal(100, 10.to_bn.sqr)
   end
 
+  def test_abs
+    assert_equal(@e1, @e2.abs)
+    assert_equal(@e3, @e4.abs)
+    assert_not_equal(@e2, @e2.abs)
+    assert_not_equal(@e4, @e4.abs)
+    assert_false(@e2.abs.negative?)
```
[the remark this answers is not in the document]
Comment (rickmark) [reply test/openssl/test_bn.rb:122]: Lol, wat?  Sure - easy fix

File: ext/openssl/ossl_bn.c (hunk 1 above)
Comment (rhenium) [reply ext/openssl/ossl_bn.c:983]: Good catch, the behavior of `OpenSSL::BN#+@` is a bug and needs to be fixed.

File: .gitignore (hunk 2 above)
Comment (rhenium) [reply .gitignore:12]: IMO these are a better fit in `~/.config/git/ignore` rather than in .gitignore of every project.

File: ext/openssl/ossl_bn.c (hunk 1 above)
Comment (rickmark) [reply ext/openssl/ossl_bn.c:983]: I'll revise the PR to correct both to returning dups of their input

**Review by rhenium — APPROVED** [review]

**Comment by rhenium** [comment]
Thank you!
